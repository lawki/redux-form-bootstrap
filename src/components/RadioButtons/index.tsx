import * as React from 'react';
import { Row, Col } from 'reactstrap';
import classnames from 'classnames';
import FormComponent from '../FormComponent';
import { BaseFieldProps } from '../../common/interfaces/CommonProps';
import { Label } from '../Label';

export const RADIO_BUTTONS_DIRECTIONS = {
    HORIZONTAL: "horizontal",
    VERTICAL: "vertical"
}
export interface RadioButtonsProps {
    options: string[];
    noLabel?: boolean;
    label?: string | { htmlFor: string, value: any }
    value?: string;
    onChange: (value: string) => void;
    direction?: "horizontal" | "vertical"
}
export default class RadioButtonsBS extends React.Component<RadioButtonsProps, any>{
    static defaultProps = {
        direction: "vertical"
    }
    render() {
        const { options, noLabel, label, onChange, value, direction, ...rest } = this.props;
        const radioButtons = options && Array.isArray(options) ?
            options.map((option, index) => (
                <Row className={classnames({ "ml-4": index != 0 && direction !== RADIO_BUTTONS_DIRECTIONS.VERTICAL })}>
                    <div key={option} >
                        <input type="radio" value={option}
                            checked={value === option}
                            {...rest}
                            onClick={() => onChange(option)}
                        />
                    </div>
                    <Label value={option} className="ml-1"
                        onClick={() => onChange(option)} />
                </Row>)) : null;
        return (
            <React.Fragment>
                {!noLabel && label ? <Label value={label} /> : null}
                {direction === RADIO_BUTTONS_DIRECTIONS.HORIZONTAL ? (
                    <Row className="ml-3">
                        {radioButtons}
                    </Row>) : (
                        <Col>
                            {radioButtons}
                        </Col>
                    )}
            </React.Fragment>
        )
    }
}

export type RadioButtonsFormProps = RadioButtonsProps & BaseFieldProps;
export class RadioButtons extends FormComponent<RadioButtonsFormProps>{
    render() {
        const { options, noLabel, label, direction, input: { value } } = this.props;
        return (
            <RadioButtonsBS
                options={options}
                noLabel={noLabel}
                label={label}
                value={value}
                onChange={this.saveToStore}
                direction={direction}
            />
        )
    }
}