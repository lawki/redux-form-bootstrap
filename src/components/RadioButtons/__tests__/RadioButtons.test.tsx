import * as React from 'react';
import { shallow } from 'enzyme';
import RadioButtons from '..';
describe('RadioButtons',()=>{
    it('renders',()=>{
        const fn = jest.fn();
        const app = shallow(<RadioButtons options={[]} onChange={fn}/>);
        expect(app).toMatchSnapshot();
    });
    it('handles onChange',()=>{
        const fn = jest.fn();
        const app = shallow(<RadioButtons options={['Red', 'Blue','Green', 'Black']} onChange={fn}/>);
        app.find('input').at(1).simulate('click');
        expect(fn.call.length).toBe(1);
    });
});