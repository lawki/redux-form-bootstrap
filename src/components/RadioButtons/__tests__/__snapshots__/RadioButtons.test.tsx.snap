// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButtons renders 1`] = `
<Fragment>
  <Label
    value=""
  />
  <Col
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <Row
      className=""
      key="Red"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <div
        key="Red"
      >
        <input
          checked={false}
          onBlur={[Function]}
          onClick={[Function]}
          type="radio"
          value="Red"
        />
      </div>
      <Label
        className="ml-1"
        onClick={[Function]}
        value="Red"
      />
    </Row>
  </Col>
</Fragment>
`;
